GitHubReactions.ps1

```diff
@@ -0,0 +1,363 @@
+
+        Gets the 'eyes' reactions for issue 157 from the Microsoft\PowerShellForGitHub project.
+
+    .EXAMPLE
+        Get-GitHubIssue -OwnerName Microsoft -RepositoryName PowerShellForGitHub -Issue 157 | Get-GitHubReaction
+
+        Gets a GitHub issue and pipe it into Get-GitHubReaction to get all the reactions for that issue.
+
+    .NOTES
+        Currently, this only supports reacting to issues. Issue comments, commit comments and PR comments will come later.
+
+    .NOTES
+        The alias parameters 'number' and 'repository_url' are so that this cmdlet composes with `Get-GitHubIssue`.
+#>
+    [CmdletBinding(
+        SupportsShouldProcess,
+        DefaultParameterSetName='Elements')]
+    [Diagnostics.CodeAnalysis.SuppressMessageAttribute("PSShouldProcess", "", Justification="Methods called within here make use of PSShouldProcess, and the switch is passed on to them inherently.")]
+    param(
+        [Parameter(Mandatory, ParameterSetName='Elements')]
+        [string] $OwnerName,
+
+        [Parameter(Mandatory, ParameterSetName='Elements')]
+        [string] $RepositoryName,
+
+        [Parameter(Mandatory, ParameterSetName='Uri', ValueFromPipelineByPropertyName)]
+        [Alias("repository_url")]
+        [string] $Uri,
+
+        [Parameter(Mandatory, ParameterSetName='Elements')]
+        [Parameter(Mandatory, ParameterSetName='Uri', ValueFromPipelineByPropertyName)]
+        [Alias("number")]
+        [int64] $Issue,
+
+        [ValidateSet('+1', '-1', 'laugh', 'confused', 'heart', 'hooray', 'rocket', 'eyes')]
+        [string] $ReactionType,
+
+        [string] $AccessToken,
+
+        [switch] $NoStatus
+    )
+
+    Write-InvocationLog
+
+    $elements = Resolve-RepositoryElements
+    $OwnerName = $elements.ownerName
+    $RepositoryName = $elements.repositoryName
+
+    $telemetryProperties = @{
+        'OwnerName' = (Get-PiiSafeString -PlainText $OwnerName)
+        'RepositoryName' = (Get-PiiSafeString -PlainText $RepositoryName)
+    }
+
+    $uriFragment = "/repos/$OwnerName/$RepositoryName/issues/$Issue/reactions"
+    if ($PSBoundParameters.ContainsKey('ReactionType')) {
+        $uriFragment += "?content=" + [Uri]::EscapeDataString($ReactionType)
+    }
+
+    $description = "Getting reactions for Issue $Issue in $RepositoryName"
+
+    $params = @{
+        'UriFragment' = $uriFragment
+        'Description' =  $description
+        'AcceptHeader' = 'application/vnd.github.squirrel-girl-preview+json'
+        'AccessToken' = $AccessToken
+        'TelemetryEventName' = $MyInvocation.MyCommand.Name
+        'TelemetryProperties' = $telemetryProperties
+        'NoStatus' = (Resolve-ParameterWithDefaultConfigurationValue -Name NoStatus -ConfigValueName DefaultNoStatus)
+    }
+
+    $result = Invoke-GHRestMethodMultipleResult @params
+
+    # Add metadata to reactions so that they compose with Remove-GitHubReaction
+    if ($result) {
+        $result | Add-Member -NotePropertyMembers @{
```

Comment:
Wouldn't you have to do 

```suggestion
        $result | ForEach-Object { Add-Member -NotePropertyMembers @{
```
for this to work?

Reply:
This is not needed. `Add-Member` accepts value by pipeline:
https://github.com/PowerShell/PowerShell/blob/05cab7fa5a3e229f2665ef9cb70fbb15b9fcc7c1/src/Microsoft.PowerShell.Commands.Utility/commands/utility/AddMember.cs#L34-L43